app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,31 @@
+class FriendshipsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @friends = current_user.friends
+    @pending_friends = current_user.pending_friends
+    @friend_requests = current_user.friend_requests
+  end
+
+  def invite
+    @friendship = current_user.friendships.new(invitee_id: params[:id])
+
+    if already_invited?(params[:id])
+      redirect_to users_path, alert: 'You already invited this User'
+    elsif @friendship.save
+      redirect_to friends_path
+    end
+  end
+
+  def accept
+    this_friendship = Friendship.where(inviter_id: params[:id], invitee_id: current_user.id)
```

Comment:
- [ ] [OPTIONAL because it was not pointed out at the previous review]Let's remember to keep ActiveRecord queries and methods in model methods, this logic should not be written directly in the controller.